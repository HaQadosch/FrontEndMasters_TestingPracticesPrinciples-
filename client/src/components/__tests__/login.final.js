import React from 'react'
import {generate} from 'til-client-test-utils'
import {render} from 'react-testing-library'
import Login from '../login'

test('calls onSubmit with the username and password when submitted', () => {
  // Arrange
  const fakeUser = generate.loginForm()
  const handleSubmit = jest.fn()
  const {getByLabelText, getByText} = render(<Login onSubmit={handleSubmit} />)

  const usernameNode = getByLabelText(/username/i)
  const passwordNode = getByLabelText(/password/i)

  // Act
  usernameNode.value = fakeUser.username
  passwordNode.value = fakeUser.password
  getByText(/submit/i).click()

  // Assert
  expect(handleSubmit).toHaveBeenCalledTimes(1)
  expect(handleSubmit).toHaveBeenCalledWith(fakeUser)
})

test('snapshot', () => {
  const {container} = render(<Login />)
  expect(container).toMatchSnapshot()
})
