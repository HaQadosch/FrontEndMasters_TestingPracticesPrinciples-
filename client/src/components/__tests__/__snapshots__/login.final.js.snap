// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  min-width: 200px;
  max-width: 400px;
  margin-left: -116px;
}

@media only screen and (max-width:819px) {
  .emotion-4 {
    margin-left: 0;
    width: 90%;
  }
}

.emotion-3 {
  display: grid;
  grid-template-columns: 100px 1fr;
  grid-gap: 16px;
  font-size: 18px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 30px;
  margin-bottom: 30px;
  width: 100%;
}

.emotion-0 {
  background: white;
  height: 50px;
  border: none;
  border-radius: 10px;
  color: var(--black);
  box-shadow: var(--shadow);
  border-bottom: 5px;
  width: 100%;
  min-width: 150px;
  display: block;
  padding-left: 10px;
}

.emotion-0::-webkit-input-placeholder {
  opacity: 0.5;
}

.emotion-0::-moz-placeholder {
  opacity: 0.5;
}

.emotion-0:-ms-input-placeholder {
  opacity: 0.5;
}

.emotion-0::placeholder {
  opacity: 0.5;
}

.emotion-2 {
  font-size: 13px;
  font-family: Raleway,sans-serif;
  background: var(--green);
  padding: 10px 20px;
  display: block;
  margin-left: auto;
  color: white;
  border: none;
  border-radius: 10px;
  box-shadow: var(--shadow);
  cursor: pointer;
  -webkit-transition: 0.5s;
  transition: 0.5s;
}

.emotion-2:hover {
  box-shadow: var(--shadowHover);
}

<div>
  <div>
    <div
      class="emotion-5"
    >
      <form
        class="emotion-4"
      >
        <div
          class="emotion-3"
        >
          <label
            for="username-input"
          >
            Username
          </label>
          <input
            class="emotion-0"
            id="username-input"
            name="username"
            placeholder="Username..."
          />
          <label
            id="password-input"
          >
            Password
          </label>
          <input
            aria-labelledby="password-input"
            class="emotion-0"
            name="password"
            placeholder="Password..."
            type="password"
          />
          <button
            class="emotion-2"
            style="margin-top: 20px;"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
